### ISSUE TITLE
html_for_tweet() has problems with partly-repeated hashtags

### ISSUE THREAD

**Issue by philgyford** [ryanmcgrath/twython#413]
If a tweet contains a hashtag that partly repeats another one, the linking of them gets confused.

eg, [this tweet](https://twitter.com/philgyfordtest/status/723184683945701376) has text like:

```#testing #test```

and its HTML should come out as:

```html
<a href=”https://twitter.com/search?q=%23testing” class=”twython-hashtag”>#testing</a> <a href=”https://twitter.com/search?q=%23test” class=”twython-hashtag”>#test</a>
```

instead it's coming out like:

```html
<a href="https://twitter.com/search?q=%23testing" class="twython-hashtag"><a href="https://twitter.com/search?q=%23test" class="twython-hashtag">#test</a>ing</a> <a href="https://twitter.com/search?q=%23test" class="twython-hashtag">#test</a>
```

i.e. the `test` is linked within `#testing`, as well as `#testing` being linked itself.

Here's JSON for the tweet from the API:

```json
{  
  "retweeted":false,
  "user":{  
    "verified":false,
    "listed_count":3,
    "default_profile":false,
    "friends_count":18,
    "created_at":"Fri Mar 23 16:56:52 +0000 2007",
    "profile_image_url_https":"https://abs.twimg.com/sticky/default_profile_images/default_profile_5_normal.png",
    "profile_background_image_url_https":"https://abs.twimg.com/images/themes/theme1/bg.png",
    "url":null,
    "id_str":"2030131",
    "profile_background_color":"9AE4E8",
    "profile_text_color":"000000",
    "lang":"en",
    "notifications":false,
    "screen_name":"philgyfordtest",
    "statuses_count":455,
    "profile_sidebar_border_color":"000000",
    "protected":false,
    "has_extended_profile":false,
    "utc_offset":3600,
    "name":"Phil Gyford Test",
    "profile_image_url":"http://abs.twimg.com/sticky/default_profile_images/default_profile_5_normal.png",
    "is_translator":false,
    "profile_use_background_image":true,
    "profile_background_image_url":"http://abs.twimg.com/images/themes/theme1/bg.png",
    "location":"",
    "profile_background_tile":false,
    "following":true,
    "id":2030131,
    "geo_enabled":true,
    "profile_link_color":"0000FF",
    "time_zone":"London",
    "entities":{  
      "description":{  
        "urls":[  

        ]
      }
    },
    "followers_count":8,
    "is_translation_enabled":false,
    "profile_sidebar_fill_color":"E0FF92",
    "follow_request_sent":false,
    "contributors_enabled":false,
    "default_profile_image":true,
    "description":"",
    "favourites_count":0
  },
  "created_at":"Thu Apr 21 16:20:39 +0000 2016",
  "is_quote_status":false,
  "id_str":"723184683945701376",
  "geo":null,
  "contributors":null,
  "text":"#testing #test",
  "lang":"und",
  "favorite_count":1,
  "place":null,
  "truncated":false,
  "coordinates":null,
  "in_reply_to_user_id":null,
  "favorited":true,
  "in_reply_to_status_id":null,
  "id":723184683945701376,
  "entities":{  
    "hashtags":[  
      {  
        "indices":[  
          0,
          8
        ],
        "text":"testing"
      },
      {  
        "indices":[  
          9,
          14
        ],
        "text":"test"
      }
    ],
    "urls":[  

    ],
    "user_mentions":[  

    ],
    "symbols":[  

    ]
  },
  "in_reply_to_status_id_str":null,
  "in_reply_to_user_id_str":null,
  "source":"<a href=\"http://tapbots.com/software/tweetbot/mac\" rel=\"nofollow\">Tweetbot for Mac</a>",
  "in_reply_to_screen_name":null,
  "retweet_count":0
}
```

**Comment by philgyford** [comment]
After spending some time puzzling over this, I've realised this [was fixed](https://github.com/ryanmcgrath/twython/commit/2ea45abb734af5a3de1e89a4ed2ea1a63e482160) eight months ago, but there hasn't been an updated release of twython since then.

**Comment by ryanmcgrath** [comment]
Wait, what? A new release should be on Pypi. If not I'll look this weekend.






On Fri, Apr 22, 2016 at 3:38 AM -0700, "Phil Gyford" <notifications@github.com> wrote:












Closed #413.

**Comment by michaelhelmick** [comment]
@ryanmcgrath I pushed a release a little bit ago. Sorry for the hold up @philgyford
